src/send-reminders.js

```diff
@@ -0,0 +1,178 @@
+import pMap from 'p-map'
+import VError from 'verror'
+import { serializeError } from 'serialize-error'
+
+let apiRoot
+let ctpClient
+let logger
+let stats
+
+async function sendReminders({
+  apiRoot: _apiRoot,
+  ctpClient: _ctpClient,
+  logger: _logger,
+}) {
+  stats = {
+    processedTemplateOrders: 0,
+    updatedTemplateOrders: 0,
+    skippedTemplateOrders: 0,
+  }
+
+  try {
+    apiRoot = _apiRoot
+    ctpClient = _ctpClient
+    logger = _logger
+
+    const activeStateId = await _fetchActiveStateId()
+
+    const orderQuery =
+      await _buildQueryOfTemplateOrdersThatIsReadyToSendReminder(activeStateId)
+
+    for await (const templateOrders of ctpClient.fetchPagesGraphQl(orderQuery))
+      await pMap(
+        templateOrders,
+        (order) => _processTemplateOrder(activeStateId, order),
+        { concurrency: 3 }
+      )
+  } catch (err) {
+    logger.error(
+      'Failed on send reminders, processing should be restarted on the next run.' +
+        `Error: ${JSON.stringify(serializeError(err))}`
+    )
+  }
+
+  return stats
+}
+
+async function _fetchActiveStateId() {
+  const {
+    body: { id },
+  } = await apiRoot.states().withKey({ key: 'Active' }).get().execute()
```

Comment:
Maybe better `commercetools-subscriptions-active` as key so that it does not conflict with potentially existing keys?

Reply:
https://github.com/commercetools/commercetools-subscriptions/pull/8/commits/930e92e2ffe4a78ccea3072e58a65671684dce5b

Reply:
https://github.com/commercetools/commercetools-subscriptions/pull/8/commits/d952cb6cc596344ce018a1f9817942d8f80f36d5